# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsPresContext.h
# Commit: a8b9260fbdef
# Full Hash: a8b9260fbdef0f3600dc0b8e2e245cb0f82c3470
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-09-08 21:39:05
# Regressor Bug: 1715501
# File Overlap Count: 1
# Description:
#   Bug 1715501 - patch 2 - Pass the presContext to platform font lookup methods so they can query it for font visibility. r=emilio
#   
#   This does not in itself change user-visible behavior, but is a foundation for bug
#   1715507 where we will make the behavior per-context instead of global. This is basically
#   just plumbing, passing a pointer to the presContext that's asking for fonts to be
# ==============================================================================

diff -r bed82cc6a625 -r a8b9260fbdef layout/base/nsPresContext.h
--- a/layout/base/nsPresContext.h	Wed Sep 08 12:18:17 2021 +0000
+++ b/layout/base/nsPresContext.h	Wed Sep 08 12:18:17 2021 +0000
@@ -163,6 +163,8 @@
 
   void UpdateFontCacheUserFonts(gfxUserFontSet* aUserFontSet);
 
+  FontVisibility GetFontVisibility() const { return mFontVisibility; }
+
   /**
    * Get the nsFontMetrics that describe the properties of
    * an nsFont.
@@ -1149,6 +1151,11 @@
   // mDynamicToolbarMaxHeight or `app units per device pixels` changes.
   void AdjustSizeForViewportUnits();
 
+  // Call in response to prefs changes that might affect what fonts should be
+  // visibile to CSS. Returns whether the current visibility value actually
+  // changed (in which case content should be reflowed).
+  bool UpdateFontVisibility();
+
   // IMPORTANT: The ownership implicit in the following member variables
   // has been explicitly checked.  If you add any members to this class,
   // please make the ownership explicit (pinkerton, scc).
@@ -1354,6 +1361,8 @@
   unsigned mInitialized : 1;
 #endif
 
+  FontVisibility mFontVisibility = FontVisibility::Unknown;
+
  protected:
   virtual ~nsPresContext();
 